subprojects/gst-plugins-base/gst-libs/gst/gl/gstglupload.c

```diff
@@ -2458,9 +2492,18 @@ gst_gl_upload_set_caps (GstGLUpload * upload, GstCaps * in_caps,
     GstCaps * out_caps)
 {
   gboolean ret;
+  GstCaps *tmp = in_caps;
 
   GST_OBJECT_LOCK (upload);
-  ret = _gst_gl_upload_set_caps_unlocked (upload, in_caps, out_caps);
+
+  if (upload->priv->method_impl &&
+      gst_egl_caps_has_feature (tmp, GST_CAPS_FEATURE_MEMORY_DMABUF)) {
+    struct DmabufUpload *dmabuf = upload->priv->method_impl;
+    dmabuf->modifier = gst_egl_caps_get_modifier (tmp);
```

Comment:
Here, I think we need some improvement. As we discuss before, when the set_caps does not specify the modifier explicitly, we need to record that. Then later we can use the legacy EGL API to let the driver find the modifier by itself.